sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -211,14 +211,58 @@ case class Subtract(left: Expression, right: Expression) extends BinaryArithmeti
   """)
 case class Multiply(left: Expression, right: Expression) extends BinaryArithmetic {
```

Comment:
Adding interval scaling here violates the BinaryArithmetic/BinaryOperator contract. We should use a different expression, and convert in the analyzer.

Reply:
I have started creating a new expression rather than making changes in the Multiply expression. Meanwhile, I am not sure how we will map "*" to "Multiply" and "MultiplyInterval" expression based on the parameter datatypes.  I can see in function Registry that multiple function names can be mapped to the same expression, but I don't see any symbol to be mapped to do different expressions based on the parameters. Can you please share an example where this is happening already?

Reply:
I also tried adding the conversion in  ResolveFunctions in Analyzer..but that didn't work. Can you please help me here.